### PR TITLE
NUI-1582: Export shared SDK code from asset-compute-pipeline

### PR DESCRIPTION
**Pull request by Ewocker**
## Description

Export shared SDK code from asset-compute-pipeline
Fixes https://jira.corp.adobe.com/browse/NUI-1582

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jdelbick** [comment]
dep mod tests are failing (for the sdk)
```
  3 failing

  1) api.js (shell)
       shellScriptWorker()
         should run a shell script with multiple renditions:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/circleci/repo/test/shellscript.test.js)
      at listOnTimeout (internal/timers.js:554:17)
      at processTimers (internal/timers.js:497:7)

  2) api.js (shell)
       shellScriptWorker()
         should catch a failing shell script:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/circleci/repo/test/shellscript.test.js)
      at listOnTimeout (internal/timers.js:554:17)
      at processTimers (internal/timers.js:497:7)

  3) api.js (shell)
       shellScriptWorker()
         should automatically set execution permissions on shell script:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/circleci/repo/test/shellscript.test.js)
      at listOnTimeout (internal/timers.js:554:17)
      at processTimers (internal/timers.js:497:7)
```

**Comment by tmathern** [comment]
Relates to https://github.com/adobe/asset-compute-pipeline/pull/24

**Comment by tmathern** [comment]
General:
I think this work here is the occasion to turn a bunch of older set of functions into classes (possibly with only static functions). This should also render the exports more structured.

File: lib/worker.js [hunk 1]
```diff
@@ -35,6 +30,7 @@ const EVENT_RENDITION_FAILED = "rendition_failed";
 
 const METRIC_RENDITION = "rendition";
 const TIMEOUT_BUFFER = 15000; // time before an action timeout when to send timeout metrics
+const IS_PIPELINE_RENDITION = false;
```
Comment (tmathern) [inline lib/worker.js:33]: Why a boolean constant for this?

File: lib/worker.js [hunk 2]
```diff
@@ -312,7 +311,12 @@ class AssetComputeWorker {
                 // at this point, we have the rendition a worker created, available at rendition.path
                 // naming rules are rendition0.extension, rendition1.extension, etc.
                 // put postprocessed rendition in a new file inside the post/ directory
-                const newRendition = new Rendition(rendition.originalInstructions, this.directories.postprocessing, rendition.index);
+                const newRendition = new Rendition(
+                    rendition.originalInstructions,
+                    this.directories.postprocessing,
+                    rendition.index,
+                    IS_PIPELINE_RENDITION
```
Comment (tmathern) [inline lib/worker.js:318]: Why a constant? (It should at least be called `NON_PIPELINE_RENDITION`).

File: test/shellscript.test.js [hunk 3]
```diff
@@ -508,14 +508,14 @@ describe("api.js (shell)", () => {
...
-            for (let i = 0; i < CMD_SIZE_LIMIT; i += 4) {
+            for (let i = 0; i < Utils.getCommandSizeLimit(); i += 4) {
                 longStringValue += (Math.random() + 1).toString(36).substring(2,6);
             }
 
             // This variable is below the size limit but only by a little bit.  It should not be written to a file
             // the other variable/values add up to over 750 characters already, so we need to chop off more than that to keep the argument from
             // exceeding the command size limit.  To accomodate for path variances, we round up to 850.
-            const shorterStringValue = longStringValue.substring(0, CMD_SIZE_LIMIT - 850);
+            const shorterStringValue = longStringValue.substring(0, Utils.getCommandSizeLimit() - 850);
```
Comment (tmathern) [inline test/shellscript.test.js:518]: You can also keep the constant `CMD_SIZE_LIMIT` and get its value only once at top of the file (avoid the repeated call).

File: lib/worker.js (hunk 1 above)
Comment (Ewocker) [reply lib/worker.js:33]: removed

File: lib/worker.js (hunk 2 above)
Comment (Ewocker) [reply lib/worker.js:318]: define `const pipeline = false` above and use it, similar is done in other place.

File: test/shellscript.test.js (hunk 3 above)
Comment (Ewocker) [reply test/shellscript.test.js:518]: done

File: lib/worker.js [hunk 4]
```diff
@@ -147,14 +142,17 @@ class AssetComputeWorker {
             this.cgroupSampler.start();
         }
 
-        this.directories = await createDirectories();
+        const folderName = '';
```
Comment (tmathern) [inline lib/worker.js:145]: (Nitpick)
```suggestion
        const folderName = "";
```

File: lib/worker.js [hunk 5]
```diff
@@ -147,14 +142,17 @@ class AssetComputeWorker {
...
+        const baseDirectory = '';
```
Comment (tmathern) [inline lib/worker.js:146]: (Nitpick)
```suggestion
        const baseDirectory = "";
```

File: lib/worker.js [hunk 6]
```diff
@@ -147,14 +142,17 @@ class AssetComputeWorker {
...
+        const pipeline = false;
```
Comment (tmathern) [inline lib/worker.js:147]: For readability, I would rename this to:
```suggestion
        const usePipeline = false;
```

File: lib/worker.js [hunk 7]
```diff
@@ -312,7 +310,13 @@ class AssetComputeWorker {
                 // at this point, we have the rendition a worker created, available at rendition.path
                 // naming rules are rendition0.extension, rendition1.extension, etc.
                 // put postprocessed rendition in a new file inside the post/ directory
-                const newRendition = new Rendition(rendition.originalInstructions, this.directories.postprocessing, rendition.index);
+                const pipeline = false;
```
Comment (tmathern) [inline lib/worker.js:313]: I would either rename to `usePipeline`, or drop the variable.

**Review by tmathern — APPROVED** [review]

**Review by jdelbick — COMMENTED** [review]
all the tests that are removed, were they moved to the pipeline?

**Comment by Ewocker** [comment]
Yes, all existing tests are moved or merged in pipeline. Added couple more for specific cases where function has conditions on pipeline vs non-pipeline, but we don't have much tests for pipeline cases originally.

**Comment by Ewocker** [comment]
> all the tests that are removed, were they moved to the pipeline?

Yes, all existing tests are moved or merged in pipeline. Added couple more for specific cases where function has conditions on pipeline vs non-pipeline, but we don't have much tests for pipeline cases originally.

**Pull request merged**